compiler/rustc_target/src/asm/xtensa.rs

```diff
@@ -0,0 +1,233 @@
+use std::fmt;
+
+use rustc_data_structures::fx::FxIndexSet;
+use rustc_span::{Symbol, sym};
+
+use super::{InlineAsmArch, InlineAsmType, ModifierInfo};
+use crate::spec::{RelocModel, Target};
+
+def_reg_class! {
+    Xtensa XtensaInlineAsmRegClass {
+        reg,
+        freg,
```

Comment:
LLVM doesn't seem to support this. `XtensaTargetLowering::getRegForInlineAsmConstraint` only handles `r` constraints.